### PR TITLE
Add snapshot hashes to gossip

### PR DESCRIPTION
**Pull request by sakridge**
#### Problem

Validators wanting to download a snapshot don't know the hash or age of snapshots created by the validators in the network, so cannot make a good decision on which one to download.

#### Summary of Changes

Add the last 24 snapshot slots and hashes to gossip to give the validator a better idea who to download from.

Fixes #

### CODE REVIEW

File: core/src/validator.rs [hunk 1]
```diff
@@ -343,6 +344,26 @@ impl Validator {
                 .set_entrypoint(entrypoint_info.clone());
         }
 
+        if let Some(ref snapshot_config) = config.snapshot_config {
```
Comment (mvines) [inline core/src/validator.rs]: This is great, but can we move the trusted_validators stuff into a separate PR.  I'm thinking of something a little different, so I'd like more time to chat about that while allowing the crds stuff to land ASAP

Comment (sakridge) [reply core/src/validator.rs]: ok
